packages/agoric-cli/src/lib/vaults.js

```diff
@@ -0,0 +1,66 @@
+// @ts-check
+
+import { COSMOS_UNIT } from './format.js';
+
+/** @typedef {import('@agoric/smart-wallet/src/offers').OfferSpec} OfferSpec */
+/** @typedef {import('@agoric/smart-wallet/src/offers').OfferStatus} OfferStatus */
+/** @typedef {import('@agoric/smart-wallet/src/smartWallet').BridgeAction} BridgeAction */
+
+/**
+ * Open a vault
+ *
+ * @param {Record<string, Brand>} brands
+ * @param {({ wantMinted: number, giveCollateral: number })} opts
+ * @returns {Proposal}
+ */
+const makeOpenProposal = (brands, opts) => {
+  return {
+    give: {
+      Collateral: {
+        brand: brands.IbcATOM,
+        value: BigInt(opts.giveCollateral * Number(COSMOS_UNIT)),
+      },
+    },
+    want: {
+      Minted: {
+        brand: brands.IST,
+        value: BigInt(opts.wantMinted * Number(COSMOS_UNIT)),
+      },
+    },
+  };
+};
+
+/**
+ * @param {Instance} instance
+ * @param {Record<string, Brand>} brands
+ * @param {{ offerId: number, wantMinted: number, giveCollateral: number }} opts
+ * @returns {BridgeAction}
+ */
+export const makeOpenSpendAction = (instance, brands, opts) => {
+  const proposal = makeOpenProposal(brands, opts);
+
+  console.warn('vaults open give', proposal.give);
+  console.warn('vaults open want', proposal.want);
+
+  // NB: not really a Proposal because the brands are not remotes
+  // Instead they're copyRecord like  "{"boardId":"board0257","iface":"Alleged: IST brand"}" to pass through the boardId
```

Comment:
I think making them remotes would be a matter of tweaking their prototype or adding a magic property... looks like `Far` or [Remotable](https://github.com/endojs/endo/blob/master/packages/pass-style/src/make-far.js#L78) would do it.

Reply:
looks like changing `harden({ [slotKey]: slot, iface })` to `Far(iface, { [slotKey]: slot, iface })` would do it:

https://github.com/Agoric/agoric-sdk/blob/c2cedde989634151fc8083ee8b45484860dc0f3a/packages/agoric-cli/src/lib/rpc.js#L193

perhaps modulo getting rid of a redundant `Alleged: `.